OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/WebViewActivity.java

```diff
@@ -66,27 +65,28 @@ public void onPagePartiallyLoaded() {
 
         webView.loadUrl(getIntent().getStringExtra(ARG_URL));
 
-        final TextView tv_dialog_title = (TextView) findViewById(R.id.tv_dialog_title);
-        final View viewSeperator = findViewById(R.id.view_seperator);
         final String title = getIntent().getStringExtra(ARG_TITLE);
-        if (TextUtils.isEmpty(title)) {
-
-            tv_dialog_title.setVisibility(View.INVISIBLE);
-            viewSeperator.setVisibility(View.INVISIBLE);
-        } else {
-            tv_dialog_title.setVisibility(View.VISIBLE);
-            viewSeperator.setVisibility(View.VISIBLE);
-            tv_dialog_title.setText(title);
+        if (!TextUtils.isEmpty(title)) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            setTitle(title);
         }
+        else {
+            getSupportActionBar().hide();
+        }
+    }
 
-        findViewById(R.id.positiveButton).setOnClickListener(new OnClickListener() {
-            public void onClick(View v) {
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item)
+    {
+        switch(item.getItemId()) {
+            case android.R.id.home:
                 finish();
```

Comment:
Why not extend from `BaseFragmentActivity`, which already implements this menu handler?